Generated Sarif files match specifications

tibbon opened the pull request:
because: The sarif files generated before were not always valid per the specificiation and would not validate.

The results property was not always an array of results. The 2.1.0 specification reads:

In all other circumstances, results SHALL be present and SHALL contain all results detected by the tool. If the tool did not detect any results, results SHALL be an empty array.

The ReportingDescriptor/helpUri property was not settable. The ReportingDescriptor/Help property was not settable.

Tests were not named with CamelCase.

Golang 1.18 was from 2022.

ToolComponent/DownloadUri was inconsitently cased as DownloadURI. ToolComponent/InformationUri was not settable.

Compiled and example files were elligible for git checkin.

https://sarifweb.azurewebsites.net/Validation
https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html

this commit: Adds support for the above. The results property is now always an array of results. Additional tests have been added.

Golang is now 1.22.1. Ran gofmt on all files.

Adds simple GitHub Actions workflow for validating tests and formatting

tibbon committed:
Generated Sarif files matching specification

because: The sarif files generated before were not always valid
per the specificiation and would not validate.

The results property was not always an array of results.
The 2.1.0 specification reads:

In all other circumstances, results SHALL be present and SHALL contain
all results detected by the tool. If the tool did not detect any
results, results SHALL be an empty array.

The ReportingDescriptor/helpUri property was not settable.
The ReportingDescriptor/Help property was not settable.

Tests were not named with CamelCase.

Golang 1.18 was from 2022.

ToolComponent/DownloadUri was inconsitently cased as DownloadURI.
ToolComponent/InformationUri was not settable.

Compiled and example files were elligible for git checkin.

https://sarifweb.azurewebsites.net/Validation
https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html

this commit: Adds support for the above. The results property is now
always an array of results. Additional tests have been added.

Golang is now 1.22.1. Ran gofmt on all files.

Adds simple GitHub Actions workflow for validating tests and formatting

dogancanbakir commented on .github/workflows/go.yml:
```diff
@@ -0,0 +1,46 @@
+# This workflow will test and format a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
```
Let's follow the same conventions (naming, file naming etc) as other projects. For example, https://github.com/projectdiscovery/httpx/blob/main/.github/workflows/build-test.yml

tibbon replied:
Updated! I reorganized the workflows to resemble the httpx project where it made sense. I've run them all in my [branch actions](https://github.com/tibbon/sarif/actions) and they pass.

tibbon committed:
Expand GitHub Actions

because: @dogancanbakir requested in a PR on the
prior commit that the naming conventions align
with other projectdiscovery projects. It seems that
adding some of the standard suite of workflows will
aid in quicker development and consistency across
projects.

this commit: Adds additional workflows in a similar
manner to the projectdiscovery/httpx project.
